### DISCUSSION TITLE
Easy One? Please advise on setting or argument to tweak for this annoying Minor Error

### DISCUSSION THREAD

**Discussion by miles-du** [oobabooga/text-generation-webui#2360] [Q&A]
Hi I often get the following error when running the more interesting 13B-4Bit quantised Models. Otherwise they work great.

 File "<torch_package_2>.multi_acc_v3_package.py", line 340, in apply_tts
    raise Exception("Model couldn't generate your text, probably it's too long")
Exception: Model couldn't generate your text, probably it's too long

It means I have to reload the Interface, it would be great if I could tweak something to stop this occuring - even if the solution is to reduce the response length.

Really would appreciate any help!

**Comment by ossirytk** [comment]
I think llama.cpp sets the default response length to 128 tokens. I'm not really sure where in the ui this is set, but you can create a custom model setting file in the presets folder and input n-predict: -1 or n-predict: 256 or something. "-1" allows the response to continue until an eos token is received so sometimes storytelling models can go on for a while. My settings file is:
`do_sample: true
ctx-size: 2048
top_p: 0.9
top_k: 40
temperature: 0.7
repetition_penalty: 1.1
typical_p: 1.0
n-predict: -1
keep: -1`
Some of these are just llama.cpp defaults. Keep "-1" keeps the initial prompt always in memory so the character card always stays in memory.

**Reply by miles-du** [reply]
Thanks, I will try this and I wasn't aware you could have a model settings file.

**Reply by ossirytk** [reply]
Honestly, Things change so fast that it's kind of hard to keep track how things work. Half of the time I'm hacking something together. However I'm not sure if this is your issue or the response text is supposed to truncate. I think the ui can truncate text in some cases, but I'm not entirely sure what causes that.

**Reply by miles-du** [reply]
I know. It's so dynamic at the moment. Now I'm re-building my Ai Box entirely, Win 11 with WSL2, Linux and see if that improves the situation, as I upgraded to a 3090 but it wasn't the game-changer I'd hoped for speed-wise. Then I will try your ideas. Thanks.

Cheers!

**Comment by ossirytk** [comment]
From what I understand, the biggest problem for gpu is the vram size. You can't really load 30B model fully into vram in a affordable commercial computer. CPU ram is cheap, so to me this seems like the best path. There are some performance upgrades ever month, so I am pretty optimistic in future cpu performance.

**Reply by miles-du** [reply]
Okay, but I just couldn't bear the response time. Even with my setup it's slow.

**Comment by Domino9752** [comment]
File "<torch_package_2>.multi_acc_v3_package.py", line 340, in apply_tts
    raise Exception("Model couldn't generate your text, probably it's too long")

 Note the "apply_tts".
 This is an error I get when I forget to reduce max_new_tokens to less than 200 when I'm using silero_tts

I've also seen the warning:
<torch_package_1>.multi_acc_v3_package.py:68: UserWarning: Text string is longer than 1000 symbols.
  warnings.warn('Text string is longer than 1000 symbols.')

1,000 characters is between 142 words and 250 words with spaces included in the character count.
Hence I limit max_new_tokens to 200 to avoid this error when using silero_tts.

BTW:  If your error is indeed from using silero_tts, just turn silero_tts off by unchecking  "Activate TTS:  in the silero_tts configuration at the bottom of the text generation. For me, everything (which was frozen after the error) starts working again.
